x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/EnterpriseSearch.java

```diff
@@ -472,8 +470,9 @@ public List<Setting<?>> getSettings() {
 
     @Override
     public List<QuerySpec<?>> getQueries() {
-        return singletonList(
-            new QuerySpec<>(RuleQueryBuilder.NAME, RuleQueryBuilder::new, p -> RuleQueryBuilder.fromXContent(p, getLicenseState()))
+        return List.of(
+            new QuerySpec<>(RuleQueryBuilder.NAME, RuleQueryBuilder::new, p -> RuleQueryBuilder.fromXContent(p, getLicenseState())),
+            new QuerySpec<>(RuleQueryBuilder.NAME_PRE_8_15, RuleQueryBuilder::new, p -> RuleQueryBuilder.fromXContent(p, getLicenseState()))
```

Comment:
This allows existing `rule_query` syntax to work alongside the new `rule` syntax.

Reply:
I wonder if we should log a deprecation when using the legacy name.

Reply:
Can we use a single `QuerySpec` with a `ParseField` since it handles deprecation?

Reply:
Oh TIL! Thanks, updating!